Create rule S6814 (#3250)
SonarSource/rspec: rules/S6814/java/rule.adoc

== Why is this an issue?

> There are two ways in Spring how to make a REST parameter optional:

Reviewer: ```suggestion
Spring provides two options to mark a REST parameter as optional:
```

1. Use `required = false` in the `@PathVariable` annotation of the respective method parameter or
2. Use type `java.util.Optional<T>` for the method parameter

> When using 1., the absence of the parameter when the REST function is called
> is encoded by `null`, which can only be used for object types.

Reviewer: ```suggestion
When using the first option, the absence of a parameter during a REST function call
is encoded by `null`, which can only be assigned to variables with an object type.
```

If `required = false` is used for a parameter with a primitive
and the REST function is called without the parameter,

> a runtime exception occurs because the Spring data mapper cannot map the `null` value to the parameter.

Reviewer: ```suggestion
a runtime exception is thrown because the Spring data mapper cannot map the `null` value to the primitive parameter.
```
